aerogear/aerogear-js-sdk
# Feat: Conflict handling feature

## Motivation

Change involve couple different things:

- Making implementation more object oriented (however there is still work on the actual link itself)
- Adding pluggable next state handler
- Adding listener for conflicts
- Cleaning up conflict resolver strategies - removing deepmerge

Decided to push that early for feedback while I'm continuing doing end to end testing and client improvements.

## Verification

1. Grab https://github.com/aerogear/apollo-voyager-ionic-example/pull/15
2. Npm link this PR (this was not published)
```
npm link . 
```
On Showcase
```
npm link @aerogear/datasync-js
```
3. Start server following instructions
4. Start client
5. Add new item. 
6. Note version is 1
7. Edit item note version changed to 2
8. Open new incognito tab with cloned version
9. Make change in both tab (notice conflict happening)

Note: that integration is currently using some simple dialog. There is
separate ticket to build good UI for this.
